client/installation-helper.cc

```diff
@@ -195,6 +195,46 @@ step4()
     cout << "done" << endl;
 }
 
+void
+step5(const string& root_prefix, const string& snapshot_type, unsigned int pre_num, const string& description, const map<string, string>& userdata)
+{
+    // fate #317973
+
+    // preconditions (maybe incomplete):
+    // snapper rpms installed
+
+    unsigned int num;
+    Snapshots::iterator snapshot;
+    SCD scd;
+
+    scd.read_only = true;
+    scd.description = description;
+    scd.userdata = userdata;
+
+    Snapper snapper("root", root_prefix);
+
+    try
+    {
+        if (snapshot_type == "single") {
+            snapshot = snapper.createSingleSnapshot(scd);
+        } else if (snapshot_type == "pre") {
+            snapshot = snapper.createPreSnapshot(scd);
+        } else if (snapshot_type == "post") {
+            Snapshots snapshots = snapper.getSnapshots();
+            Snapshots::iterator pre = snapshots.find(pre_num);
+            snapshot = snapper.createPostSnapshot(pre, scd);
+        }
+    }
+    catch (const runtime_error& e)
+    {
+        y2err("create snapshot failed, " << e.what());
+        exit(EXIT_FAILURE);
+    }
+
+    cout << snapshot->getNum() << endl;
+    exit(EXIT_SUCCESS);
```

Comment:
The other stepX functions return instead of calling exit. Using exit in C++ will not call all destructors, http://en.cppreference.com/w/cpp/utility/program/exit.